zapcore/console_encoder.go

```diff
@@ -64,8 +63,8 @@ func (c consoleEncoder) Clone() Encoder {
 	return consoleEncoder{c.jsonEncoder.Clone().(*jsonEncoder)}
 }
 
-func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) ([]byte, error) {
-	line := bytes.NewBuffer(buffers.Get())
+func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) (*buffer.Buffer, error) {
```

Comment:
Primary advantage is less concrete: it more cleanly separates encoders, which turn objects into bytes, and Facilities/WriteSyncers. It also preserves some degree of API similarity with the marshaler types in the standard library, which typically return `([]byte, error)`.

If you're staunchly against this, though, I can certainly switch.